client/app/components/Paginator.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import AntPagination from 'antd/lib/pagination';
+
+export function Pagination({
+  page,
+  itemsPerPage,
+  totalCount,
+  onChange,
+}) {
+  if (totalCount <= itemsPerPage) {
+    return null;
+  }
+  return (
+    <div className="paginator-container">
+      <AntPagination
+        className="pagination"
+        defaultCurrent={page}
+        defaultPageSize={itemsPerPage}
+        total={totalCount}
+        onChange={onChange}
+      />
+    </div>
+  );
+}
+
+Pagination.propTypes = {
+  page: PropTypes.number,
+  itemsPerPage: PropTypes.number,
+  totalCount: PropTypes.number,
+  onChange: PropTypes.func,
+};
```

Comment:
The Angular version of the Paginator component took as input a Paginator model object. This saved a lot of repeated code where it was used. Why not do the same with the React version?

Reply:
Probably a better idea. I'm not sure what it will take to use the Pagination model there. Two other alternatives are:

1. Make `<dynamic-table>` have the same "interface" as the pagination model.
2. Have another component that uses the one you have here, that takes the model.

Reply:
It will not work - `react2angular` does no detect changes of object properties, so we cannot pass model object here because everywhere it's populated after the first render

Reply:
We can have a simple wrapper in Angular that passes the properties from the object to React, and once all the Pagination component users are in React get rid of it. This is an isolated change that will be easier to cleanup than bad API used everywhere.

Reply:
We will need to cleanup bad API anyways. Currently we have a lot of places where we use Paginator from the react templates; we can create wrapper to pass model objects, but later we will move all that pages to React, and there we will use Paginator component directly with its bad API - so I see no profit.

Reply:
No, the idea is that once we move to React we keep using the object API. Basically we will start with:

1. React component that takes 4 properties.
2. Angular component that uses the React one and accepts an object.

Then when we start moving tables to React, we add a React component that accepts an object. When all using code is in React, we can remove the Angular middleware.

This is all isolated changes in a single place, instead of tracking down all the places that use this component. Also, if we had done this from the beginning, you wouldn't have to repeat the $applyAsync code and wouldn't have discrepancies ;-)

As a side note, considering how often the issue of passing an object comes up -- maybe we can make it work somehow?